### PR TITLE
[RemoveDIs][DebugInfo] Update SROA to handle DPVAssigns

### PR DESCRIPTION
**Pull request by SLTozer**
SROA needs to update llvm.dbg.assign intrinsics when it migrates debug info in response to alloca splitting; this patch updates the debug info migration code to handle DPVAssigns as well, making use of generic code to avoid duplication as much as possible.

### CODE REVIEW

File: llvm/lib/Transforms/Scalar/SROA.cpp [hunk 1]
```diff
@@ -319,6 +319,29 @@ static DebugVariable getAggregateVariable(DbgVariableIntrinsic *DVI) {
   return DebugVariable(DVI->getVariable(), std::nullopt,
                        DVI->getDebugLoc().getInlinedAt());
 }
+static DebugVariable getAggregateVariable(DPValue *DPV) {
+  return DebugVariable(DPV->getVariable(), std::nullopt,
+                       DPV->getDebugLoc().getInlinedAt());
+}
+
+static DPValue *createLinkedAssign(DPValue *, DIBuilder &DIB,
```
Comment (OCHyams) [inline llvm/lib/Transforms/Scalar/SROA.cpp:327]: clang-format?

**Review by OCHyams — APPROVED** [review]
Code changes all LGTM, but this needs tests too. What's your intended strategy for landing code/test changes?

FWIW For DPDeclares I updated the tests for each pass in the same patch the code was update (e.g., you would update SROA tests here), with the tests "rotten green" (not actually testing the DPDeclares) until the capstone patch that enabled dbg.declare -> DPDeclare landed.

**Review by OCHyams — CHANGES_REQUESTED** [review]
(oops wrong button)

**Review by OCHyams — APPROVED** [review]
LGTM (after clang-format)

**Comment by cHackz18** [comment]
@SLTozer Hi Stephen,

it looks as though your change may have caused a build bot failure here:
https://lab.llvm.org/buildbot/#/builders/139/builds/57800

**Comment by SLTozer** [comment]
Apologies, I see the fix has already been landed - thanks for catching this!

**Pull request merged**